### DISCUSSION TITLE
Any way to get citation keys into ios/ipados zotero or cloud library?

### DISCUSSION THREAD

**Discussion by paultopia** [retorquere/zotero-better-bibtex#2996] [Q&A]
I'm dealing with a weird workflow issue with BBT which might be just me, but there's a *chance* someone else has dealt with this and solved it, so asking here just to see...

The tl'dr is that I do a lot of writing from the road on an ipad (LTE access FTW). The ios/ipados version of zotero is usable enough for me that I can add sources from there, and I just write in a github repo. BUT: if I add a source to zotero from the ipad, I don't have the BBT cite key until I get back to the real computer to sync that client and then get a CSL-JSON file.  This is an annoying interruption to my workflow---my files tend to have a lot of terrible placeholders of the form "**CITE SO-AND-SO HERE** for a long time. 

I've been pondering trying one of a variety of potential over-engineered fixes to this, like running a headless version of desktop zotero on a cloud server somewhere and triggering a sync and CSL-JSON file export with a git hook, or running some kind of script on my real computer to periodically wake up and sync zotero to trigger a new CSL-JSON build.  but... is there an easier way, like any kind of direct access hidden away somewhere to citation keys from the ipad client or the web library or something?  

Thanks!

**Comment by retorquere** [comment]
I don't know how Zotero works on iPads but if you can influence the items being created and are willing to generate the keys yourself you can just stick them into the `extra` field and you'd be set.

Barring that, if you want to use BBT; the citation keys don't exist anywhere until BBT generates them locally, and they exist only local to the instance of Zotero where BBT is creating them. You can sync them by (auto)pinning them because pinning writes them into the `extra` field so they become part of the item.

Running Zotero headless is easy enough on Linux (, and you can automate absolutely everything in it using the [debug bridge](https://github.com/retorquere/zotero-better-bibtex/tree/master/test/fixtures/debug-bridge) (you can ignore the parts about vscode, it's just a REST API you can call into).
